kalite/topic_tools/content_models.py

```diff
@@ -395,11 +394,21 @@ def annotate_content_models(db=None, channel="khan", language="en", ids=None, **
     Our ids can be duplicated at the moment, so this may be several content items per id.
     When a content item has been updated, propagate availability up the topic tree.
     """
-    content_models = get_content_items(ids=ids, channel=channel, language=language)
-    updates_dict = update_content_availability(content_models)
+    def iterator_content_items(ids=None, **kwargs):
+        if ids:
+            items = Item.select().where(Item.id.in_(ids)).dicts().iterator()
+        else:
+            items = Item.select().dicts().iterator()
 
-    with Using(db, [Item]):
+        mapped_items = itertools.imap(unparse_model_data, items)
+
+        updated_mapped_items = update_content_availability(mapped_items)
```

Comment:
As mentioned verbally, we should modify ```update_content_availability``` to be a generator too, otherwise this will just cause inflation again.